src/TwigComponent/src/ComponentTemplateFinder.php

```diff
@@ -15,33 +15,53 @@
 
 /**
  * @author Matheo Daninos <matheo.daninos@gmail.com>
+ *
+ * @internal
  */
 final class ComponentTemplateFinder implements ComponentTemplateFinderInterface
 {
+    private readonly ?string $directory;
+
     public function __construct(
-        private Environment $environment
+        private Environment $environment,
+        string $directory = null,
     ) {
+        $this->directory = \is_string($directory) ? (rtrim($directory, '/').'/') : null;
```

Comment:
We should trigger a deprecation is `$directory` is null, right? We want to force that to be passed in the future?

Reply:
We cannot add an "@internal" on the constructor i guess ? So yes :(